lib/resty/balancer/utils.lua

```diff
@@ -43,9 +26,22 @@ do
             end
             return count
         end
+
+    else
+        jit_nkeys = nkeys
     end
 end
-
 _M.nkeys = nkeys
 
+
+function _M.copy(nodes)
+    local newnodes = new_tab(0, jit_nkeys and jit_nkeys(nodes) or 4)
+    for id, weight in pairs(nodes) do
```

Comment:
Note that this loop cannot be JIT compiled.

Reply:
Does anything need to change? we can not avoid `pairs` here.

Reply:
If it's not on the hot code path, then it's fine.